src/main/java/pl/javastart/streamstask/StreamsTask.java

```diff
@@ -32,28 +31,41 @@ private void run() {
 
         Collection<User> women = findWomen(users);
         Double averageMenAge = averageMenAge(users);
-        Map<Long, List<Expense>> expensesByUserId = groupExpensesByUserId(users, expenses);
+        Map<Long, List<Expense>> expensesByUserId = groupExpensesByUserId(expenses);
         Map<User, List<Expense>> expensesByUser = groupExpensesByUser(users, expenses);
     }
 
     // metoda powinna zwracać listę kobiet (sprawdzając, czy imię kończy się na "a")
     Collection<User> findWomen(Collection<User> users) {
-        throw new RuntimeException("Not implemented");
+        return users.stream().filter(user -> user.getName().endsWith("a")).toList();
     }
 
     // metoda powinna zwracać średni wiek mężczyzn (sprawdzając, czy imię nie kończy się na "a")
     Double averageMenAge(Collection<User> users) {
-        throw new RuntimeException("Not implemented");
+        BigDecimal maleUsersNumber = new BigDecimal(users.stream().filter(user -> !user.getName().endsWith("a")).count());
+        return users.stream()
+                .filter(user -> !user.getName().endsWith("a"))
+                .mapToInt(User::getAge)
+                .mapToObj(BigDecimal::new)
+                .reduce(BigDecimal::add)
+                .map(bd -> bd.divide(maleUsersNumber, 2, RoundingMode.HALF_UP))
+                .map(bigDecimal -> Double.parseDouble(bigDecimal.toString())).get();
```

Comment:
Tutaj sugeruję użycie mapToDouble, a jeżeli zależy Ci na zaokrągleniu to użyj np. klasy NumberFormat